### PR TITLE
create campaign function integration with frontend  

### PR DESCRIPTION
**Pull request by 0xshikhar**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
     "next/typescript"
   ],
   "rules": {
+    "no-unused-vars": "warn",
```
Comment (positonic) [inline .eslintrc.json:7]: Why are we not rather keeping the code clean than disabling best practices?

File: components/create-campaign.tsx [hunk 2]
```diff
@@ -26,56 +27,49 @@ export function CreateCampaign() {
...
-          title: formData.title,
-          description: formData.description,
-          fundingGoal: parseEther(formData.fundingGoal || '0'),
-          startTime: BigInt(new Date(formData.startTime).getTime() / 1000),
-          endTime: BigInt(new Date(formData.endTime).getTime() / 1000)
-        }
-      ]
-    })
+    // Generate a unique identifier hash for the campaign
+    const identifierHash = keccak256(
```
Comment (positonic) [inline components/create-campaign.tsx:40]: Why do we need to create our own identifier? Doesn't CCP create one for us?

File: components/create-campaign.tsx [hunk 3]
```diff
@@ -26,56 +27,49 @@ export function CreateCampaign() {
...
+    )
+
+    const campaignData = {
+      launchTime: BigInt(new Date(formData.startTime).getTime() / 1000),
+      deadline: BigInt(new Date(formData.endTime).getTime() / 1000),
+      goalAmount: parseEther(formData.fundingGoal || '0'),
+    }
+
+    // using array instead of object to maintain order
+    const campaignData1 = [
```
Comment (positonic) [inline components/create-campaign.tsx:51]: This needs a better variable name

File: components/create-campaign.tsx [hunk 4]
```diff
@@ -26,56 +27,49 @@ export function CreateCampaign() {
...
-      await writeContract({
+      const result = await writeContract({
         address: campaignInfoFactory as `0x${string}`,
         abi: CampaignInfoFactoryABI,
         functionName: 'createCampaign',
         args: [
-          address as `0x${string}`,
-          ('0x' + '1'.repeat(64)) as `0x${string}`,
+          address,
+          identifierHash,
```
Comment (positonic) [inline components/create-campaign.tsx:64]: I see we are sending a unique Id to CCP. I imagine this should be our unique Akashic identifier?

File: lib/test.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// function createCampaign(bytes32 platformBytes) internal {
```
Comment (positonic) [inline lib/test.js:1]: Can we rather stash this and commit it when we're ready?

**Review by positonic — CHANGES_REQUESTED** [review]
I've left some feedback on the PR look forward to your responses

File: .eslintrc.json (hunk 1 above)
Comment (0xshikhar) [reply .eslintrc.json:7]: yes, we are following them - here instead of error, just converted them into warnings"

Comment (positonic) [reply .eslintrc.json]: Let's keep them as errors. Prefer to comment out the code if necessary, but we should be deleting such code IMO.

**Review by positonic — CHANGES_REQUESTED** [review]

**Pull request merged**